### PR TITLE
nexd: Add proxy config validation

### PR DESCRIPTION
**Pull request by russellb**
commit 74e74d23ac257be5b3d5ba9781027870744f3325
Author: Russell Bryant <russell.bryant@gmail.com>
Date:   Thu Apr 20 14:08:47 2023 -0400

    nexd: Add proxy config validation
    
    Do validation on proxy rules at startup. Add an e2e test that
    validates various error cases.
    
    Signed-off-by: Russell Bryant <rbryant@redhat.com>

commit 39aea4b2729b3db7b39474ee8b759965d8cd5da5
Author: Russell Bryant <russell.bryant@gmail.com>
Date:   Thu Apr 20 14:07:23 2023 -0400

    nexd: Reorder build steps to improve layer caching
    
    I noticed that every build was re-running `go mod download`. This
    reordering seems to get it to keep that portion cached.
    
    Signed-off-by: Russell Bryant <rbryant@redhat.com>

### CODE REVIEW

File: integration-tests/proxy_test.go [hunk 1]
```diff
@@ -437,3 +437,106 @@ func TestProxyIngressAndEgress(t *testing.T) {
...
+		"--egress", "tcpa:8080:100.100.0.1:80"})
+	require.Error(err)
+
+	// Incomplete ingress rule
+	_, err = helper.containerExec(ctx, node1, []string{"nexd", "--username", username, "--password", password, "proxy",
+		"--ingress", "tcp:8080"})
+	require.Error(err)
+
+	// Incomplete egress rule
+	_, err = helper.containerExec(ctx, node1, []string{"nexd", "--username", username, "--password", password, "proxy",
```
Comment (vishnoianil) [inline integration-tests/proxy_test.go:537]: I think taking out this string array (given it's exactly same parameter) and declare it at the start of the test would clean the containerExec across the test ?

File: internal/nexodus/userspace_proxy.go [hunk 2]
```diff
@@ -100,6 +100,23 @@ func (ax *Nexodus) UserspaceProxyAdd(ctx context.Context, wg *sync.WaitGroup, pr
 	if err != nil {
 		return err
 	}
+	if port < 1 || port > 65535 {
```
Comment (vishnoianil) [inline internal/nexodus/userspace_proxy.go:103]: i think the conditions should be && and not ||?

File: internal/nexodus/userspace_proxy.go [hunk 3]
```diff
@@ -111,6 +128,9 @@ func (ax *Nexodus) UserspaceProxyAdd(ctx context.Context, wg *sync.WaitGroup, pr
 	if err != nil {
 		return err
 	}
+	if destPort < 1 || destPort > 65535 {
```
Comment (vishnoianil) [inline internal/nexodus/userspace_proxy.go:131]: || -> &&

**Review by vishnoianil — CHANGES_REQUESTED** [review]

File: internal/nexodus/userspace_proxy.go (hunk 2 above)
Comment (vishnoianil) [reply internal/nexodus/userspace_proxy.go:103]: Nevermind.

File: integration-tests/proxy_test.go (hunk 1 above)
Comment (russellb) [reply integration-tests/proxy_test.go]: done, I put the common args in a new `baseArgs` slice at the beginning

**Review by vishnoianil — APPROVED** [review]

**Pull request merged**